### PR TITLE
feat: support add custom headers for etcd requests

### PR DESCRIPTION
**Pull request by tzssangglass**
Signed-off-by: tzssangglass <tzssangglass@gmail.com>

This PR supports adding custom headers to etcd requests on the v3 protocol, used in scenarios where etcd is being proxied and so on.

### CODE REVIEW

**Review by membphis — APPROVED** [review]
LGTM

**Review by nic-chen — COMMENTED** [review]
better add a test case for header `Authorization`, to avoid subsequent changes that cause problems here

**Comment by tzssangglass** [comment]
> better add a test case for header `Authorization`, to avoid subsequent changes that cause problems here

update

File: lib/resty/etcd/v3.lua [hunk 1]
```diff
@@ -125,6 +126,15 @@ local function _request_uri(self, method, uri, opts, timeout, ignore_auth)
         end
     end
 
+    if self.extra_headers and type(self.extra_headers) == "table" then
+        for key, value in pairs(self.extra_headers) do
+            if key ~= "Authorization" then
```
Comment (tokers) [inline lib/resty/etcd/v3.lua:131]: Setting for Content-Length, Transfer-Encoding, Connection, Upgrade should be ignored.

**Review by nic-chen — APPROVED** [review]

File: t/v3/key.t [hunk 2]
```diff
@@ -284,7 +284,8 @@ timeout/
 --- timeout: 5
 
 
-=== TEST 4.1: watchdir(key=="")
+
+=== TEST 6.1: watchdir(key=="")
```
Comment (tokers) [inline t/v3/key.t:288]: What use a period? Just `TEST 6` is OK.

File: lib/resty/etcd/v3.lua (hunk 1 above)
Comment (tzssangglass) [reply lib/resty/etcd/v3.lua]: update

File: t/v3/key.t (hunk 2 above)
Comment (tzssangglass) [reply t/v3/key.t]: update

**Review by tokers — APPROVED** [review]

**Pull request merged**